### ISSUE TITLE
How to Feed Batched Sequences of Images through Tensorflow conv2d

### ISSUE THREAD

**Issue by RylanSchaeffer** [tensorflow/tensorflow#19896]
This seems like a trivial question, but I've been unable to find the answer. Maybe I'm just bad at Googling, but I think the question is sufficiently common to justify an easily accessible answer.

I have batched sequences of images of shape:

`[batch_size, number_of_frames, frame_height, frame_width, number_of_channels]`

and I would like to pass each frame through a few convolutional and pooling layers. However, `tf.layers.conv2d` accepts 4D inputs of shape:

`[batch_size, frame_height, frame_width, number_of_channels]`

My first attempt was to use tf.map_fn over axis=1, but I discovered that [this function does not propagate gradients](https://github.com/tensorflow/tensorflow/issues/3972) - which isn't mentioned in the [documentation](https://www.tensorflow.org/api_docs/python/tf/map_fn) (but should be!).

My second attempt was to use `tf.unstack` over the first dimension and then use `tf.while_loop`. However, my batch_size and number_of_frames are dynamically determined (i.e. both are None), and `tf.unstack` raises `{ValueError} Cannot infer num from shape (?, ?, 30, 30, 3) if num is unspecified`. I tried specifying `num=tf.shape(observations)[1]`, but this raises `{TypeError} Expected int for argument 'num' not <tf.Tensor 'A2C/infer/strided_slice:0' shape=() dtype=int32>`.

What is the proper way to feed batched sequences of images through `conv2d`?

### System information
OS Platform and Distribution: Ubuntu 16.04.4 LTS
TensorFlow installed from binary.
TensorFlow version: 'v1.8.0-2159-gea9fb80'
Python version: 2.7.12

**Comment by drpngx** [comment]
You could use `tf.contrib.recurrent.Recurrent` to instantiate a range for loop.

**Comment by RylanSchaeffer** [comment]
This seems strange. I want to apply the same function to each batch of images for each step in the sequence - I shouldn't need any recurrency in my network for this computation. Is there no alternative way?

**Comment by RylanSchaeffer** [comment]
My roommate just pointed out a trivial solution: reshape into 

`[batch_size * number_of_frames, frame_height, frame_width, number_of_channels]`

and then reshape after.

**Comment by facaiy** [comment]
How about using conv3d whose kernel is like [1,…… ]?
